packages/shorebird_cli/lib/src/shorebird_flutter_manager.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/process.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+
+/// {@template shorebird_flutter_manager}
+/// Helps manage the Flutter installation used by Shorebird.
+/// {@endtemplate}
+class ShorebirdFlutterManager {
+  /// {@macro shorebird_flutter_manager}
+  const ShorebirdFlutterManager();
+
+  static const String flutterGitUrl =
+      'https://github.com/shorebirdtech/flutter.git';
+
+  Future<void> installRevision({required String revision}) async {
+    final targetDirectory = Directory(
+      p.join(shorebirdEnv.flutterDirectory.parent.path, revision),
+    );
+    if (targetDirectory.existsSync()) return;
+    const executable = 'git';
+
+    // Clone the Shorebird Flutter repo into the target directory.
+    final cloneArgs = [
+      'clone',
+      '--filter=tree:0',
+      flutterGitUrl,
+      '--no-checkout',
+      targetDirectory.path,
+    ];
```

Comment:
Thoughts on removing this from the install script (not in this PR, obviously)

Reply:
yeah was thinking about that as well 👍

Reply:
https://github.com/shorebirdtech/shorebird/issues/1014